code/game/machinery/dish_drive.dm

```diff
@@ -91,7 +91,7 @@
 				step_towards(I, src)
 
 /obj/machinery/dish_drive/AltClick(mob/living/user)
-	if(user.canUseTopic(src))
+	if(user.Adjacent(src))
```

Comment:
Apologies for having been a bit vague
`if(!user.canUseTopic(src, !issilicon(user)))` would be more appropriate here. 
canUseTopic checks stat / incapacitated / ect as needed for the mob, and the second arg handles adjacency ( in this case, `!issilicon(user)`, which means they have to be adjacent if they are not a silicon.) 
If you used just Adjacent, any mob/living could use the dish_drive as long as they are nearby regardless if they are dead or not (this including butterflies/crabs/other simple mobs/ect.)